haystack/retriever/dense.py

```diff
@@ -74,17 +75,18 @@ def __init__(self,
         self.remove_sep_tok_from_untitled_passages = remove_sep_tok_from_untitled_passages
 
         # Init & Load Encoders
-        self.query_tokenizer = DPRQuestionEncoderTokenizer.from_pretrained(query_embedding_model)
+        self.query_tokenizer = AutoTokenizer.from_pretrained(query_embedding_model)
```

Comment:
I check for Question it was correctly picked `DPRQuestionEncoderTokenizer`, but for passage if `bart-large` is used then it pick `BARTEncoderTokenizer` instead of `DPRPassageEncoderTokenizer`. `DPRPassageEncoderTokenizer` only work on selected model, hence I think better to use `AutoTokenizer` of transformers and allow user to override if they need some customisation

Reply:
:+1:

Reply:
BTW reverted the change to minimize the change on already working codebase. Currently without this change also I managed generator to work.